src/avram/save_operation.cr

```diff
@@ -153,7 +153,7 @@ abstract class Avram::SaveOperation(T)
     def after_prepare
       validate_required *required_attributes
 
-      {% if primary_key_type == :uuid %}
+      {% if primary_key_type.id == UUID.id %}
         id.value ||= UUID.random()
```

Comment:
I wonder if this should be handled on the postgres side?

Reply:
I kinda think so, but it does require an extension to be enabled. Maybe that is fine? I'd much rather have it on the Postgres side if possible. I'll open another issue for it though

Reply:
https://github.com/luckyframework/avram/issues/122